CMakeLists.txt

```diff
@@ -44,10 +46,13 @@ set(warehouse_srcs
   src/mongo_ros_dummy.cpp)
 
 add_library(warehouse_ros SHARED ${warehouse_srcs})
-
 target_link_libraries(warehouse_ros ${catkin_LIBRARIES} ${MongoDB_LIBRARIES} ${OPENSSL_LIBRARIES} ${Boost_LIBRARIES})
-catkin_add_gtest(test_mongo_roscpp test/test_mongo_ros.cpp)
-target_link_libraries(test_mongo_roscpp warehouse_ros)
+
+if (CATKIN_ENABLE_TESTING)
+  catkin_add_gtest(test_mongo_roscpp test/test_mongo_ros.cpp)
+  target_link_libraries(test_mongo_roscpp warehouse_ros)
+  add_rostest(test/mongo_ros.test)
```

Comment:
You should either add the executable as a gtest or register a rostest calling the executable. Both is likely not desired?